contracts/exchange/test/balance_stores/local_balance_store.ts

```diff
@@ -0,0 +1,106 @@
+import { assetDataUtils } from '@0x/order-utils';
+import { AssetProxyId } from '@0x/types';
+import { BigNumber } from '@0x/utils';
+import * as _ from 'lodash';
+
+import { BalanceStore } from './balance_store';
+
+export class LocalBalanceStore extends BalanceStore {
+    /**
+     * Creates a new balance store based on an existing one.
+     * @param balanceStore Existing balance store whose values should be copied.
+     */
+    public static create(sourceBalanceStore?: BalanceStore): LocalBalanceStore {
```

Comment:
Why not just have this behavior in the constructor?

Reply:
Hmm I guess it seems awkward to me to include an instance of a class as a parameter to its own constructor. I guess we could parameterize the raw balance struct, although I'm hoping to abstract away the inner mechanics as much as possible.